server/login.go

```diff
@@ -44,14 +44,15 @@ func GetLogin(c *gin.Context) {
 		getLoginOauth2(c)
 	}
 
+	login := ToUser(c)
+
 	// exit if authorization fails
-	// TODO(bradrydzewski) return an error message instead
 	if c.Writer.Status() != 200 {
+		log.Errorf("could not get authorization for %s.", login.Login)
+		c.Redirect(303, "/login#error=access_denied")
 		return
 	}
 
-	login := ToUser(c)
```

Comment:
ToUser(c) will not contain a user object if the login fails. So `failedAuthorization := ToUser(c)` would still cause a panic in a failure scenario. Also, no need to redirect since the below functions would have already redirected:

```
	switch {
	case settings.Service.OAuth == nil:
		getLoginBasic(c)
	case settings.Service.OAuth.RequestToken != "":
		getLoginOauth1(c)
	default:
		getLoginOauth2(c)
	}
```